### ISSUE TITLE
n_warmup + n_optimize stop-gradient split for long transient leads

### ISSUE THREAD

**Issue by bk-squared** [bk-squared/rfx#40]
Plan §5.2 item from issue #31. Skip gradient accumulation through the warmup portion of the scan for memory/compute savings on long transient runs.

### Idea
```python
sim.forward(n_warmup=5000, n_optimize=2000, ...)
```
- First 5000 steps: `jax.lax.stop_gradient` on outputs, runs as a plain scan (no AD tape).
- Last 2000 steps: full AD tape (or segmented).

Reduces memory when the pulse takes many steps to settle before the measurement window of interest.
